Modules/clinic/itertoolsmodule.c.h

```diff
@@ -206,8 +206,8 @@ itertools__grouper(PyTypeObject *type, PyObject *args, PyObject *kwargs)
     if (!_PyArg_CheckPositional("_grouper", PyTuple_GET_SIZE(args), 2, 2)) {
         goto exit;
     }
-    if (!PyObject_TypeCheck(PyTuple_GET_ITEM(args, 0), &groupby_type)) {
-        _PyArg_BadArgument("_grouper", "argument 1", (&groupby_type)->tp_name, PyTuple_GET_ITEM(args, 0));
+    if (!PyObject_TypeCheck(PyTuple_GET_ITEM(args, 0), clinic_state()->groupby_type)) {
+        _PyArg_BadArgument("_grouper", "argument 1", (clinic_state()->groupby_type)->tp_name, PyTuple_GET_ITEM(args, 0));
```

Comment:
With #101477, this becomes a lot faster; we should add a `get_module_state_by_cls` and use that instead of `clinic_state` for the `subclass_of=` in `itertools._grouper.__new__` clinic, so we can easily benefit from the new fast API when that PR has landed.

Reply:
I'll update the PR so it is ready for such a change.

Reply:
Added; resolving.